sqlmesh/core/engine_adapter/base.py

```diff
@@ -481,15 +489,82 @@ def _create_table_from_columns(
             if primary_key and self.SUPPORTS_INDEXES
             else []
         )
-        schema = exp.Schema(
-            this=table,
-            expressions=[
-                exp.ColumnDef(this=exp.to_identifier(column), kind=kind)
-                for column, kind in columns_to_types.items()
-            ]
-            + primary_key_expression,
+
+        column_descriptions = kwargs.pop("column_descriptions", None)
+        schema = self._build_schema_exp(
+            table,
+            columns_to_types,
+            column_descriptions if self.COMMENT_CREATION == CommentCreation.IN_SCHEMA_DEF else None,
+            primary_key_expression,
+        )
+
+        table_description = kwargs.pop("table_description", None)
+        self._create_table(
+            schema,
+            None,
+            exists=exists,
+            columns_to_types=columns_to_types,
+            table_description=table_description
+            if self.COMMENT_CREATION == CommentCreation.IN_SCHEMA_DEF
```

Comment:
I'd probably do this check closer to the source to make sure we don't mess this logic up in other places where we call `_create_table`. So let's move this into `_create_table`.